packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -524,6 +525,7 @@ function updateSimpleMemoComponent(
     if (
       shallowEqual(prevProps, nextProps) &&
       current.ref === workInProgress.ref &&
+      (current.effectTag & ForceUpdateForLegacySuspense) === NoEffect &&
```

Comment:
I would move this check into the block below, and it it's true, set `didReceiveUpdate` to true to prevent a later bailout.